provisioner/node_pools.go

```diff
@@ -219,62 +456,13 @@ func (p *AWSNodePoolProvisioner) Reconcile(ctx context.Context, updater updatest
 	return nil
 }
 
-// prepareCloudInit prepares the user data by rendering the golang template.
-// It also uploads the dynamically generated files needed for the nodes in the pool
-// A EC2 UserData ready base64 string will be returned.
-func (p *AWSNodePoolProvisioner) prepareCloudInit(nodePool *api.NodePool, values map[string]interface{}) (string, error) {
-	var (
-		keyName, poolKind string
-	)
-	if nodePool.IsMaster() {
-		keyName = masterFilesKMSKey
-		poolKind = "master"
-	} else {
-		keyName = workerFilesKMSKey
-		poolKind = "worker"
-	}
-
-	kmsKey, err := getPKIKMSKey(p.awsAdapter, p.cluster.LocalID, keyName)
-	if err != nil {
-		return "", err
-	}
-
-	renderer := &FilesRenderer{
-		awsAdapter: p.awsAdapter,
-		cluster:    p.cluster,
-		config:     p.config,
-		directory:  poolKind,
-		nodePool:   nodePool,
-	}
-
-	s3Path, err := renderer.RenderAndUploadFiles(values, p.bucketName, kmsKey)
-	if err != nil {
-		return "", err
-	}
-
-	p.logger.Debugf("Uploaded generated files to %s", s3Path)
-	values[s3GeneratedFilesPathValuesKey] = s3Path
-
-	cloudInitContents, err := p.config.NodePoolManifest(nodePool.Profile, cloudInitFileName)
-	if err != nil {
-		return "", err
-	}
-	rendered, err := renderSingleTemplate(cloudInitContents, p.cluster, nodePool, values, p.awsAdapter)
-	if err != nil {
-		return "", err
-	}
-	var gzipBuffer bytes.Buffer
-	writer := gzip.NewWriter(&gzipBuffer)
-	defer writer.Close()
-	_, err = writer.Write([]byte(rendered))
-	if err != nil {
-		return "", err
-	}
-	err = writer.Close()
-	if err != nil {
-		return "", err
+func supportsT2Unlimited(instanceTypes []string) bool {
```

Comment:
Can you add a comment about this? I vaguely remember a comment about it, but looking at it now I have no clue why we would need this function? :)